float32[] my_pos
float32[] friend_pos
bool ns
bool emergency
bool team
float32 time
int32[] friend_action
int32[] action_list
int32[] cup_color
int32 cup
int32[] hand
float32[] enemy1_pos
float32[] enemy2_pos
int32 strategy
---
int32[] action
float32[] position
int32[] cup
